### PR TITLE
fuse2

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Review by xnox — CHANGES_REQUESTED** [review]
needs checking what to do with /dev/fuse

File: fuse2.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  - name: ${{package.name}}-doc
+    pipeline:
+      - uses: split/manpages
+    description: fuse manpages
+
+  - name: ${{package.name}}-static
+    pipeline:
+      - uses: split/static
+    description: fuse static
```
Comment (xnox) [inline fuse2.yaml:74]: Sadly ordering matters. `split/static` must come before `split/dev` as both split the static libraries.
In this lexical ordering -dev package ends up having static libraries and static package is empty.

```
$ tar -tf fuse2-dev-*.apk 2>/dev/null | grep usr/lib
usr/lib
usr/lib/libfuse.a
usr/lib/libfuse.so
usr/lib/libulockmgr.a
usr/lib/libulockmgr.so
usr/lib/pkgconfig
usr/lib/pkgconfig/fuse.pc

$ tar -tf fuse2-static-*.apk 2>/dev/null
.SIGN.RSA.local-melange.rsa.pub
.PKGINFO
var
var/lib
var/lib/db
var/lib/db/sbom
var/lib/db/sbom/fuse2-static-2.9.9-r0.spdx.json
```

File: s3fs-fuse.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  description: FUSE-based file system backed by Amazon S3
+  copyright:
+    - license: GPL-2.0-only
+  # https://github.com/s3fs-fuse/s3fs-fuse/blob/master/COMPILATION.md#note---the-required-librariescomponents-required-to-run-s3fs-are
+  dependencies:
+    runtime:
+      - fuse2
+      - openssl
+      - curl
+      - libxml2
```
Comment (xnox) [inline s3fs-fuse.yaml:14]: I expect all of these to have soname depends; which are better than fixed depends. As soname depends will remain correct during transitions. Please remove these.

File: s3fs-fuse.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      - busybox
+      - ca-certificates-bundle
+      - curl-dev
+      - fuse2-dev
+      - gcc
+      - gettext-dev
+      - git
+      - libtool
+      - libxml2-dev
+      - openssl-dev
```
Comment (xnox) [inline s3fs-fuse.yaml:32]: many of the above are redundant, i.e. git is pulled it automatically, and so is gcc, ca-certificates-bundle. It helps to have smaller list of things that are unique to this build. But that's a nitpick, doesn't have any material impact as all of these are installed anyway one way or another.

**Comment by xnox** [comment]
> needs checking what to do with /dev/fuse

this comment was about this

```
2024/08/20 11:08:22 INFO mknod /home/build/melange-out/fuse2/dev/fuse -m 0666 c 10 229 || true
2024/08/20 11:08:22 WARN mknod: /home/build/melange-out/fuse2/dev/fuse: Operation not permitted
```

To ensure that this either fails, or if successful is removed and not shipped in the package. End result is good that it is not shipped.

File: curl.yaml [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+  epoch: 2
   description: "URL retrieval utility and library"
   copyright:
     - license: MIT
+  dependencies:
+    runtime:
+      - brotli-dev
+      - libpsl-dev
+      - nghttp2-dev
+      - openssl-dev
```
Comment (xnox) [inline curl.yaml:13]: these must be under curl-dev subpackage!!!!!!

**Review by xnox — CHANGES_REQUESTED** [review]

File: curl.yaml (hunk 4 above)
Comment (developer-guy) [reply curl.yaml:13]: ah gotcha! I missed that part from your commit, thanks!

**Review by xnox — APPROVED** [review]

**Pull request merged**